### PR TITLE
LRDOCS-3931, forms and workflow reorganization

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LRDOCS-3931 After being asked for an intro, Charlles and I decided we wanted to reorganize the forms and workflow docs into separate top level sections. The content you should review is the introduction to the Forms section at discover/portal/articles/04-forms/00-intro.markdown. Thanks Renato!

### CODE REVIEW

File: discover/portal/articles/04-business-productivity/00-intro.markdown [hunk 1]
```diff
@@ -1,3 +1,68 @@
...
+When you need a form in @product@, what you really need is data. @product@ has
+two applications for building forms to collect precisely the data you need:
+
+1.  [Liferay Forms](/discover/portal/-/knowledge_base/7-0/collecting-information-from-users): @product@'s primary form building application lets you
+    build forms ranging from the simplest one or two question survey to the most
+    complex, multi-page, homeowners insurance application form featuring form
+    rules and lists populated by a REST data provider.
+
+2.  [Dynamic Data
+    Lists(DDL)](/discover/portal/-/knowledge_base/7-0/creating-simple-applications):
```
Comment (natocesarrego) [inline discover/portal/articles/04-business-productivity/00-intro.markdown]: Hey @rbohl,

I think is missing a blank character before **(DDL)**.

File: discover/portal/articles/04-business-productivity/00-intro.markdown [hunk 2]
```diff
@@ -1,3 +1,68 @@
...
+    Provides a user interface tool for building reusable form- and list-based
+    applications intended for display on pages in @product@. It's the ability to
+    mold the display using [templates](/discover/portal/-/knowledge_base/7-0/using-templates-to-display-forms-and-lists) that sets DDL apart.
+
++$$$
+
+**Kaleo Forms:** If you're a Liferay Digital Enterprise customer, there's a
+third form building tool in @product@ called [Kaleo
```
Comment (natocesarrego) [inline discover/portal/articles/04-business-productivity/00-intro.markdown]: @rbohl, if the Kaleo Forms can only be used by EE customers, then we should remove the @product@. 😉

File: discover/portal/articles/04-business-productivity/00-intro.markdown [hunk 3]
```diff
@@ -1,3 +1,68 @@
...
+Forms](/discover/portal/-/knowledge_base/7-0/workflow-forms). It integrates form
+building with workflow to create form based business processes, like a
+Conference Room Checkout Form, or a Support Ticket Process so support tickets go
+through the proper channels on their way to resolution. Read more about Kaleo
+Forms in the [workflow
+section](/discover/portal/-/knowledge_base/7-0/using-workflow).
+
+$$$
+
+## Which Form Builder Should I Use?
```
Comment (natocesarrego) [inline discover/portal/articles/04-business-productivity/00-intro.markdown]: I've loved this section idea! Congrats!

File: discover/portal/articles/04-business-productivity/00-intro.markdown [hunk 4]
```diff
@@ -1,3 +1,68 @@
...
+Liferay Forms gives you a lot of control over how your form is displayed,
+allowing you to make [form
+pages](/discover/portal/-/knowledge_base/7-0/creating-advanced-forms#creating-form-pages),
+and adjust the [form
+layout](/discover/portal/-/knowledge_base/7-0/creating-advanced-forms#form-layouts)
+of the fields. These styling elements are enough for most forms. Over time, more
+features will be added to Liferay Forms so that DDL might someday become
+obsolete, in favor of a single form building application, Liferay Forms. If
+you're a heavy user of DDL, don't worry, Liferay is committed to preserving and
+enhancing the features provided by DDL; they just might be moved to Liferay Forms. Now is the time to familiarize yourself with Liferay Forms and
```
Comment (natocesarrego) [inline discover/portal/articles/04-business-productivity/00-intro.markdown]: I've become a little bit uncomfortable with this sentence, once, from what I know, we won't improve DDL, actually, the idea is deprecate it. I'm just worried about create an incorrect expectation from the customers side, you know?

Comment (rbohl) [reply discover/portal/articles/04-business-productivity/00-intro.markdown]: What I meant by this part is that if customers are heavy DDL users, the features will be preserved even when DDL goes away for good. That's true, isn't it? I'll try to reword it to make this more clear. @natocesarrego

File: discover/portal/articles/04-business-productivity/00-intro.markdown (hunk 2 above)
Comment (rbohl) [reply discover/portal/articles/04-business-productivity/00-intro.markdown]: yes, good catch

File: discover/portal/articles/04-business-productivity/00-intro.markdown (hunk 1 above)
Comment (rbohl) [reply discover/portal/articles/04-business-productivity/00-intro.markdown]: You're correct. Good catch.

File: discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown [hunk 5]
```diff
@@ -2,16 +2,16 @@
 
 Collecting information from your site's Users is important, whether you're
 asking them to submit a brief survey or apply for a mortgage. In either case,
-you need to design a form. Liferay gives you almost limitless form building
-capability with the Forms application.
+you need a form. Liferay gives you almost limitless form building capability
+with the Forms application.
```
Comment (natocesarrego) [inline discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown]: Hey @rbohl, in the Forms and Lists intro, you've referenced the Forms Portlet as Liferay Forms, but here you're calling it as just Forms, thus I think that would be nice to include a note or just a sentence in the intro telling the reader that the Forms is known as Liferay Forms too. I'm raising this concern, because as we've several forms, the user can make confusion and be in doubt if we want the refer to the Forms application or all Liferay Forms, for example.

What do you think about that?

Comment (rbohl) [reply discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown]: @natocesarrego I agree. Maybe I'll just use "Liferay Forms" here too. I'll look at it in context and see what makes sense.

File: discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown [hunk 6]
```diff
@@ -25,9 +25,8 @@ needed information. Here's some of the what the Forms app can do right now
   value (not entered if left alone by user) for each field
 - Validate fields using a number of different criteria
 
-If you only need a very basic form, don't worry. Despite this seemingly long
-list of more complex options, developing a simple, elegant form to suit your
-more basic needs can be done with little effort. The next article will cover
-basic form building, and will be followed by a more complete, complex tutorial
-on the Forms app.
+If you only need a very basic form, don't worry. Despite this long list of more
```
Comment (natocesarrego) [inline discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown:29]: I've liked this! This quick intro before the next article is very nice. 👏 👏

File: discover/portal/articles/04-forms/00-intro.markdown [hunk 7]
```diff
@@ -18,23 +18,28 @@ two applications for building forms to collect precisely the data you need:
...
 **Kaleo Forms:** If you're a Liferay Digital Enterprise customer, there's a
 third form building tool in @product@ called [Kaleo
-Forms](/discover/portal/-/knowledge_base/7-0/workflow-forms). It integrates form
+Forms](https://customer.liferay.com/documentation/7.0/admin/-/official_documentation/portal/workflow-forms). It integrates form
 building with workflow to create form based business processes, like a
 Conference Room Checkout Form, or a Support Ticket Process so support tickets go
 through the proper channels on their way to resolution. Read more about Kaleo
 Forms in the [workflow
-section](/discover/portal/-/knowledge_base/7-0/using-workflow).
+section](https://customer.liferay.com/documentation/7.0/admin/-/official_documentation/portal/workflow-forms).
```
Comment (natocesarrego) [inline discover/portal/articles/04-forms/00-intro.markdown]: Have you changed your mind and have decided to refer the **workflow-forms** section instead of the **using-workflow** section?

**Comment by natocesarrego** [comment]
Hi @rbohl,

I've just finished to review this pull. Please take a look on my comments and if you need anything else, let me know.

Thanks.

File: discover/portal/articles/04-business-productivity/00-intro.markdown (hunk 4 above)
Comment (natocesarrego) [reply discover/portal/articles/04-business-productivity/00-intro.markdown]: I think it's true, but as this is something that will happen in the future, I'm not sure if ALL features from DDL will be available. Maybe we can double check it with Charlles or the PT.

What do you think?

File: discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown (hunk 5 above)
Comment (natocesarrego) [reply discover/portal/articles/04-business-productivity/01-forms/00-forms-intro.markdown]: Ok, Russell. Thanks!

File: discover/portal/articles/04-forms/00-intro.markdown (hunk 7 above)
Comment (rbohl) [reply discover/portal/articles/04-forms/00-intro.markdown]: Yes, I think it's better to use an absolute link to the customer portal here. Community users will understand they can't see the material because they are not customers, instead of being confused by an Article Not Found message, since the Kaleo Forms article isn't on LDN.

**Comment by rbohl** [comment]
Thanks @natocesarrego I'll make the review edits today.

File: discover/portal/articles/04-forms/00-intro.markdown (hunk 7 above)
Comment (natocesarrego) [reply discover/portal/articles/04-forms/00-intro.markdown]: I see your concern and it's valuable! The goal of my previous comment was just ask if you've intentionally changed the article from [https://dev.liferay.com/discover/portal/-/knowledge_base/7-0/using-workflow](https://dev.liferay.com/discover/portal/-/knowledge_base/7-0/using-workflow) to [https://customer.liferay.com/documentation/7.0/admin/-/official_documentation/portal/workflow-forms](https://customer.liferay.com/documentation/7.0/admin/-/official_documentation/portal/workflow-forms), i.e. not just change the place where the content is published, but also the referenced article.

Comment (rbohl) [reply discover/portal/articles/04-forms/00-intro.markdown]: Oh I see what you mean, @natocesarrego . Yes I did intend to link directly to Kaleo Forms instead of the intro to the workflow section.
